### PR TITLE
Hcap 696 final showdown

### PR DESCRIPTION
**Pull request by dbayly-freshworks**
Fixed HCAP-715 - Issue where the incorrect key and shape were used to check if a user was MoH in the SiteParticipants table. 

Improved the function of  HCAP-695 
- When navigating between tabs on the participant table, if you are on a page with a page number that is greater than the page total for the tab you are navigating to, you will be shown the first  page of that tab. The number will still be wrong as I haven't figured out why the component isn't updating. This also causes the page switch feature to respond improperly ( The component is set to the wrong page, so it wont update correctly until the user uses the controls to update it to the correct page. 

Fixed HCAP-696 

When a MoH withdraws a hired participant, their participant statuses were not being updated. 

Now when a MoH worker withdraws a user, a new participant status is added for each EOI. The previous status is stored in that data field and if that user's interest is switched back to 'yes', the previous status is restored. A record of this switch will exist in the database. 

Updated the Participants Table to show archived users to MoH

### CODE REVIEW

File: server/services/participants.js [hunk 1]
```diff
@@ -133,6 +134,20 @@ const updateParticipant = async (participantInfo) => {
     },
     { history: participantInfo.history, userUpdatedAt: new Date().toJSON() }
   );
+  if(changes.interested !== undefined){
```
Comment (fwkendall) [inline server/services/participants.js:137]: nitpick - the spacing here is inconsistent. I thought our linting should catch this - any idea if it didn't run / isn't looking for this?

Comment (dbayly-freshworks) [reply server/services/participants.js:137]: For whatever reason my linting doesn't run on commit and I forgot to run the formatter. Good catch!

File: server/services/participants-helper.js [hunk 2]
```diff
@@ -56,6 +56,34 @@ const addDistanceToParticipantFields = (raw, siteDistanceJoin) =>
     }),
   }));
 
+// Find the previous status for each org, create a copy of it 
+const  revertToOldStatus = async ( participantStatuses,setParticipantStatus)=>{
```
Comment (fwkendall) [inline server/services/participants-helper.js:60]: I think this is actually a change in the existing behaviour, but that's fine as long as we document it (I think before the participant would always go back to "prospecting" - my understanding here is they would go back to any prior status at all, although I believe the button is disabled for withdrawn archived participants so going back to "hired" would never be possible)

File: server/services/participants-helper.js [hunk 3]
```diff
@@ -56,6 +56,34 @@ const addDistanceToParticipantFields = (raw, siteDistanceJoin) =>
...
+}
+// Withdraw the participant 
+const insertWithdrawalParticipantStatus = async (participantStatuses,setParticipantStatus) =>{ 
+  participantStatuses.forEach( async (status)=>{
+    // Prevent locking the user into a loop of archived statuses
+    if(status.status && status.status ==='archived'){
+      return;
+    }else{
+      await setParticipantStatus(status.employer_id,status.participant_id, 'archived',{
+        final_status: 'Withdrawn by MoH',
```
Comment (fwkendall) [inline server/services/participants-helper.js:78]: Should we make this more generic? Thinking we'll probably reuse this method when participants can withdraw their own interest. Maybe just "Withdrawn from HCAP"?

File: client/src/pages/private/SiteParticipantsTable.js [hunk 4]
```diff
@@ -31,9 +31,9 @@ export default ({ siteId, onArchiveParticipantAction }) => {
   const { auth } = AuthContext.useAuth();
   const { openToast } = useToast();
   const roles = auth.user?.roles || [];
-  const isHA = checkPermissions(roles, ['health_authority']);
+  const isHA = roles.includes('health_authority');
   if (!isHA) {
-    columns = columns.filter((col) => col.id !== 'archived');
+    columns = columns.filter((col) => col.id !== 'archive');
```
Comment (fwkendall) [inline client/src/pages/private/SiteParticipantsTable.js:36]: gross, we really need a constants file somewhere to prevent this kind of thing

**Review by fwpushan — COMMENTED** [review]
LGTM for me. Please check some minor comments. And also check some action is failing.

**Review by Zenen-freshworks — COMMENTED** [review]
Yep, looks like we already have good suggestions on this PR - I have nothing to add.

File: server/services/participants-helper.js (hunk 2 above)
Comment (dbayly-freshworks) [reply server/services/participants-helper.js]: It's pretty clear this was a case that wasn't thought of when it was first implemented. 

The previous function was to set the status of the `participant` to withdrawn but not update any `participants_status`, hence the bug.  I added this part to ensure that the ability to re-engage a user after they're withdrawn by a MoH user was kept.

This all might warrant further discussion.

Comment (fwkendall) [reply server/services/participants-helper.js]: Discussed offline - MOH undoing a withdrawal is not a real use case. For now we are documenting this behaviour but it should be disabled in future.

**Review by fwkendall — APPROVED** [review]

**Pull request merged**